crates/bevy_ecs/src/archetype.rs

```diff
@@ -739,6 +755,14 @@ impl Archetypes {
     }
 }
 
+impl Index<RangeFrom<ArchetypeGeneration>> for Archetypes {
```

Comment:
Would we need to support the other range variants (Range, RangeInclusive, etc.)?

Reply:
im inclined to not add things unless they are needed, this range impl seems to be the only really useful one

Reply:
Yeah. `ArchetypeGeneration` only makes sense as a marker for the start of a range of archetypes to update, nothing else. So adding more impls would be erroneous and misleading, the API is quite precise on what it does.